cpp/scenario/gazebo/include/scenario/gazebo/Model.h

```diff
@@ -0,0 +1,222 @@
+
+    // ==========
+    // Model Core
+    // ==========
+
+    bool valid() const;
+
+    size_t dofs() const;
+    std::string name() const;
+
+    size_t nrOfLinks() const;
+    size_t nrOfJoints() const;
+
+    LinkPtr getLink(const std::string& linkName) const;
+    JointPtr getJoint(const std::string& jointName) const;
+
+    std::vector<std::string> linkNames(const bool scoped = false) const;
+    std::vector<std::string> jointNames(const bool scoped = false) const;
+
+    double controllerPeriod() const;
+    bool setControllerPeriod(const double period);
+
+    // ========
+    // Contacts
+    // ========
+
+    bool contactsEnabled() const;
+    bool enableContacts(const bool enable);
+
+    bool selfCollisions() const;
+    bool enableSelfCollisions(const bool enable);
+
+    std::vector<std::string> linksInContact() const;
+
+    std::vector<base::ContactData>
+    contacts(const std::vector<std::string>& linkNames = {}) const;
+
+    // ==================
+    // Vectorized Methods
+    // ==================
+
+    std::vector<double> jointForces( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointPositions( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointVelocities( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    bool setJointControlMode(const base::JointControlMode& mode,
+                             const std::vector<std::string>& jointNames = {});
+
+    std::vector<LinkPtr> links( //
+        const std::vector<std::string>& linkNames = {}) const;
+    std::vector<JointPtr> joints( //
+        const std::vector<std::string>& jointNames = {}) const;
+
+    // =========================
+    // Vectorized Target Methods
+    // =========================
+
+    bool setJointForces( //
+        const std::vector<double> forces,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointPositionTargets( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointVelocityTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+    bool setJointAccelerationTargets( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    bool resetJointPositions( //
+        const std::vector<double> positions,
+        const std::vector<std::string>& jointNames = {});
+    bool resetJointVelocities( //
+        const std::vector<double> velocities,
+        const std::vector<std::string>& jointNames = {});
+
+    std::vector<double> jointPositionTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointVelocityTargets( //
+        const std::vector<std::string>& jointNames = {}) const;
+    std::vector<double> jointAccelerationTargets( //
```

Comment:
This methods assume that the passed joints have 1 dof, right?

Reply:
No, the joint serialization takes into account also multi DoF joints. If the caller wants to use these vectorized methods, they should know the DoF of the joints.

This means that if you have `{'joint1_1dof', 'joint2_3dof', 'joint3_1dof'}`, the method will return a 5 elements vector.

Reply:
Ah cool.